base/math.jl

```diff
@@ -70,11 +71,23 @@ end
 
 # evaluate p[1] + x * (p[2] + x * (....)), i.e. a polynomial via Horner's rule
 macro horner(x, p...)
+    @gensym val
```

Comment:
Yes, good point. I think this was an issue in an earlier version of Julia, but in 0.6 converts it into a unique name.